event_model/__init__.py

```diff
@@ -688,6 +680,41 @@ def __call__(self, name, doc, validate=False):
         return super().__call__(name, doc, validate)
 
 
+def _attempt_with_retries(func, args, kwargs,
+                          retry_intervals,
+                          error_to_catch, error_to_raise):
+    """
+    Return func(*args, **kwargs), using a retry loop.
+
+    func, args, kwargs: self-explanatory
+    retry_intervals: list
+        How long to wait (seconds) between before each successive retry.
+    error_to_catch: Exception class
+        If this is raised, retry.
+    error_to_raise: Exception instance or class
+        If we run out of retries, raise this from the proximate error.
+    """
+    # We are sure to attempt IO at least once and
+    # then perhaps additional times depending on the contents of
+    # retry_intervals.
+    error = None
+    for interval in [0] + retry_intervals:
```

Comment:
Those would all raise a `TypeError`. (At least they would fail unambiguously!) We normalize the input to a list at its source, in `Filler.__init__`. I intentionally did not pay that price again here, since the function is internal and always passed that normalized value. I added a notice to the docstring above in 652c4aabc0c58dbbde0da691d00bdf99ad1267ef. Do you feel that is sufficient?

Reply:
Actually, maybe this function should take in the whole series, including the `0`, to avoid paying for `0 + retry_intervals` each time.

Reply:
New approach just pushed ensures via a `property` that `self.retry_intervals` is always a list and then passes `[0] + retry_intervals` into `_attempt_with_retries`, which now accepts any interable sequence. I this makes the tests easier to follow as well.

Reply:
Great, and I like it more!